compiler/rustc_mir_transform/src/gvn.rs

```diff
@@ -1405,6 +1405,29 @@ impl<'body, 'tcx> VnState<'body, 'tcx> {
             }
         }
 
+        // Aggregate-then-Transmute can just transmute the original field value,
+        // so long as the type is transparent over only that one single field.
+        if let Transmute = kind
+            && let Value::Aggregate(
+                AggregateTy::Def(aggregate_did, aggregate_args),
+                FIRST_VARIANT,
+                field_values,
+            ) = self.get(value)
+            && field_values.len() == 1
+            && let adt = self.tcx.adt_def(aggregate_did)
+            && adt.is_struct()
+            && adt.repr().transparent()
+        {
+            let field_ty = adt.non_enum_variant().single_field().ty(self.tcx, aggregate_args);
+            let field_value = field_values[0];
```

Comment:
```suggestion
            && let [field_value] = field_values[..]
            && let adt = self.tcx.adt_def(aggregate_did)
            && adt.is_struct()
            && adt.repr().transparent()
        {
            let field_ty = adt.non_enum_variant().single_field().ty(self.tcx, aggregate_args);
```
Small drive-by review, IMO this looks a little nicer with a slice pattern 🦄